src/components/Trade/TradeInput.tsx

```diff
@@ -181,84 +170,20 @@ export const TradeInput = () => {
   const hasValidSellAmount = bnOrZero(sellAmountCryptoPrecision).gt(0)
 
   const handleInputChange = useCallback(
-    async (action: TradeAmountInputField, amount: string) => {
+    (action: TradeAmountInputField, amount: string) => {
       updateAction(action)
       // If we've overridden the input we are no longer in sendMax mode
       updateIsSendMax(false)
       updateAmount(amount)
 
-      if (isSwapperApiPending && !quoteAvailableForCurrentAssetPair) {
-        await setTradeAmountsRefetchData({ amount, action })
-      } else {
-        setTradeAmountsUsingExistingData({ amount, action })
-      }
+      handleInputAmountChange()
     },
-    [
-      updateAction,
-      updateIsSendMax,
-      updateAmount,
-      isSwapperApiPending,
-      quoteAvailableForCurrentAssetPair,
-      setTradeAmountsRefetchData,
-      setTradeAmountsUsingExistingData,
-    ],
+    [updateAction, updateIsSendMax, updateAmount, handleInputAmountChange],
   )
 
-  const handleToggle = useCallback(() => {
-    try {
-      const currentValues = Object.freeze({
-        buyAsset,
-        sellAsset,
-        sellAssetFiatRate,
-        buyAssetFiatRate,
-      })
-      const currentSellAsset = sellAsset
-      const currentBuyAsset = buyAsset
-      if (!(currentSellAsset && currentBuyAsset)) return
-
-      updateSellAsset(currentBuyAsset)
-      updateBuyAsset(currentSellAsset)
-      updateSellAmountCryptoPrecision('0')
-      updateBuyAmountCryptoPrecision('0')
-      updateBuyAssetFiatRate(currentValues.sellAssetFiatRate)
-      updateSellAssetFiatRate(currentValues.buyAssetFiatRate)
-
-      // The below values all change on asset change. Clear them so no inaccurate data is shown in the UI.
-      updateFiatSellAmount('0')
-      updateFiatBuyAmount('0')
-      updateFeeAssetFiatRate(undefined)
-      updateFees(undefined)
-      updateTrade(undefined)
-      updateSelectedSellAssetAccountId(undefined)
-      updateSelectedBuyAssetAccountId(undefined)
-      updateBuyAssetAccountId(undefined)
-      updateSellAssetAccountId(undefined)
-    } catch (e) {
-      moduleLogger.error(e, 'handleToggle error')
-    }
-  }, [
-    buyAsset,
-    sellAsset,
-    sellAssetFiatRate,
-    buyAssetFiatRate,
-    updateSellAsset,
-    updateBuyAsset,
-    updateSellAmountCryptoPrecision,
-    updateBuyAmountCryptoPrecision,
-    updateBuyAssetFiatRate,
-    updateSellAssetFiatRate,
-    updateFiatSellAmount,
-    updateFiatBuyAmount,
-    updateFeeAssetFiatRate,
-    updateFees,
-    updateTrade,
-    updateSelectedSellAssetAccountId,
-    updateSelectedBuyAssetAccountId,
-    updateBuyAssetAccountId,
-    updateSellAssetAccountId,
-  ])
+  const handleToggle = useCallback(() => handleAssetToggle(), [handleAssetToggle])
```

Comment:
just an alias/relic now?

Reply:
Indeed, addressed in https://github.com/shapeshift/web/pull/4138/commits/dddbde80b84b26edfea966547554a69f2d10f290.